k8s-plugin/src/test/kotlin/com/amazon/spinnaker/keel/tests/GitArtifactSupplierTest.kt

```diff
@@ -0,0 +1,151 @@
+package com.amazon.spinnaker.keel.tests
+
+import com.amazon.spinnaker.keel.k8s.artifactSupplier.GitArtifactSupplier
+import com.amazon.spinnaker.keel.k8s.model.GitRepoArtifact
+import com.amazon.spinnaker.keel.k8s.model.GitVersion
+import com.amazon.spinnaker.keel.k8s.service.IgorArtifactServiceSupplier
+import com.fasterxml.jackson.databind.jsontype.NamedType
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.artifacts.GitMetadata
+import com.netflix.spinnaker.keel.api.artifacts.PublishedArtifact
+import com.netflix.spinnaker.keel.api.artifacts.Repo
+import com.netflix.spinnaker.keel.artifacts.DockerArtifact
+import com.netflix.spinnaker.keel.serialization.configuredYamlMapper
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.coEvery
+import io.mockk.mockk
+import strikt.api.expectThat
+import strikt.assertions.*
+
+internal class GitArtifactSupplierTest : JUnit5Minutests {
+    private val yamlMapper = configuredYamlMapper()
+    private val igorArtifactService = mockk<IgorArtifactServiceSupplier>()
+    private val gitVersion1 = GitVersion(
+        "testRepo",
+        "testProject",
+        "igor",
+        "1.2.3",
+        "123",
+        "gitHub",
+        "https://repo.url",
+        "https://repo.url/commit/123",
+        "some date",
+        "author1",
+        "message1",
+        "email@email.email"
+    )
+    private val gitVersion2 = gitVersion1.copy(version = "1.2.4", commitId = "1234")
+    private val gitVersion3 = gitVersion1.copy(version = "1.2.5", commitId = "12345")
+
+    private val invalidGitVersion1 = gitVersion1.copy(version = "v1.2.5")
+    private val invalidGitVersion2 = gitVersion1.copy(version = "v.1.2.6")
```

Comment:
looks like we have strict expectations on what the versioning strategy should look like. can we make sure we document it in the README?